src/setuptools_scm/git.py

```diff
@@ -171,3 +171,64 @@ def _git_parse_describe(describe_output):
     tag, number, node = describe_output.rsplit("-", 2)
     number = int(number)
     return tag, number, node, dirty
+
+
+def is_git_dir(d):
```

Comment:
please note that the from_potential_worktree helper in the git worktree already finds the root using git directly (this works correctly on windows and in special situations on windows)

Reply:
Great, so I should just replace
```python
if is_git_dir(dirname):
```
with 
```python
GitWorkdir.from_potential_worktree(dirname) is not None:
```
and get rid of the `is_git_dir` helper?

Reply:
there is a samefile check that needs to ensure the parent searching is working correctly, that needs to be conditional

Reply:
So I use `GitWorkdir.from_potential_worktree`, but I don't like needed the `try: ... except: ...`